practical/matlab/idr0021_steps.m

```diff
@@ -0,0 +1,203 @@
+        channelId = channel.getId().getValue();
+        lc = channel.getLogicalChannel();
+        channelName = lc.getName().getValue();
+        if channelName == target
+            channelIndex = j;
+            break;
+        end
+    end
+    
+    fprintf('Analyse Image: %s, Channel: %i\n', image.getName().getValue(), channelIndex);
+    plane = getPlane(session, image, z, channelIndex - 1, t); % channel index in OMERO starts with 0
+    threshNstd = 6;
+    minPixelsPerCentriole = 20;   % minimum size of objects of interest
+    vals = reshape(plane, [numel(plane), 1]);   % reshape to 1 column
+    mean1 = mean(vals);
+    std1 = std(vals);
+    % images are mostly background, so estimate threshold using basic stats
+    thresh1 = mean1 + threshNstd * std1;
+    bwRaw = imbinarize(plane, thresh1);
+    BWfinal = bwareaopen(bwRaw, minPixelsPerCentriole);
+    
+    [B,L] = bwboundaries(BWfinal, 'noholes');
+    roi = omero.model.RoiI;
+    max_area = 0;
+    csv_row = java.util.ArrayList;
+    for b = 1:length(B)
+        boundary = B{b};
+        x_coordinates = boundary(:,2);
+        y_coordinates = boundary(:,1);
+        shape = createPolygon(x_coordinates, y_coordinates);
+        setShapeCoordinates(shape, z, channelIndex - 1, t);
+        roi.addShape(shape);
+        area = polyarea(x_coordinates, y_coordinates);
+        max_area = max(max_area, area);
+    end
+    % Link the roi and the image
+    imageId = image.getId().getValue();
+    roi.setImage(omero.model.ImageI(imageId, false));
+    if ~isempty(B)
+        roi = iUpdate.saveAndReturnObject(roi);
+        csv_row.add(imageId);
+        csv_row.add(max_area);
+        csv_data.add(csv_row);
+    end 
+end
+
+% Create a CSV file
+headers = 'ImageID,Area';
+fileID = fopen('results.csv','w');
+fprintf(fileID,'%s\n',headers);
+for kk = 0: csv_data.size()-1
+        csv_row = csv_data.get(kk);
+        row = strcat(num2str(csv_row.get(0)), ',', num2str(csv_row.get(1)));
+        fprintf(fileID,'%s\n',row);
+end
+fclose(fileID);
+% Create and link the CSV file annotation
+fileAnnotation = writeFileAnnotation(session, 'results.csv', 'mimetype', 'text/csv', 'namespace', 'training.demo');
+linkAnnotation(session, fileAnnotation, 'dataset', datasetId);
+
+
+% Exercise 7:
+% Save the results as OMERO.table.
+% After this step go back to OMERO.web,
+% select an image and expand the 'Tables' tab
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+columns = javaArray('omero.grid.Column', 2);
+columns(1) = omero.grid.LongColumn('Image', '', []);
+columns(2) = omero.grid.DoubleColumn('Area', '', []);
+table = session.sharedResources().newTable(1, char('from_matlab'));
+table.initialize(columns);
+for kk = 0: csv_data.size()-1
+    csv_row = csv_data.get(kk);
+    row = javaArray('omero.grid.Column', 1);
+    row(1) = omero.grid.LongColumn('Image', '', [csv_row.get(0)]);
+    row(2) = omero.grid.DoubleColumn('Area', '', [csv_row.get(1)]);
+    table.addData(row);
```

Comment:
The same as for the CSV - it is a bit underwhelming to have just one single parameter per image.